firewood/src/merkle.rs

```diff
@@ -1418,4 +1354,57 @@ mod tests {
             assert_eq!(fetched_val.as_deref(), val.as_slice().into());
         }
     }
+
+    #[test]
+    fn remove_one() {
+        let key = b"hello";
+        let val = b"world";
+
+        let mut merkle = create_test_merkle();
+        let root = merkle.init_root().unwrap();
+
+        merkle.insert(key, val.to_vec(), root).unwrap();
+
+        assert_eq!(
+            merkle.get(key, root).unwrap().as_deref(),
+            val.as_slice().into()
+        );
+
+        let removed_val = merkle.remove(key, root).unwrap();
+        assert_eq!(removed_val.as_deref(), val.as_slice().into());
+
+        let fetched_val = merkle.get(key, root).unwrap();
+        assert!(fetched_val.is_none());
+    }
+
+    #[test]
+    fn remove_many() {
+        let mut merkle = create_test_merkle();
+        let root = merkle.init_root().unwrap();
+
+        // insert values
+        for key_val in u8::MIN..=u8::MAX {
+            let key = vec![key_val];
```

Comment:
nit:
```suggestion
            let key = &[key_val];
```
Then change the usages to just `key` instead of `&key`.